luaui/Widgets/gui_gridmenu.lua

```diff
@@ -508,13 +521,8 @@ local function gridmenuKeyHandler(_, _, args, _, isRepeat)
 			return
 		end
 
-		local uDef = UnitDefs[-uDefID]
-		local isRepeatMex = uDef.customParams.metal_extractor
-			and uDef.name == activeCmd
-			and not (uDef.stealth or #uDef.weapons > 0)
-		local cmd = isRepeatMex and "areamex" or spGetCmdDescIndex(uDefID)
-		Spring.SetActiveCommand(cmd, 3, false, true, alt, ctrl, meta, shift)
```

Comment:
You replaced this with `Spring.SetActiveCommand(cmd, 1, false, true,Spring.GetModKeyState())`, are there any negative implications?

Reply:
None that I have found, I tested everything I could think of. It was weird that it was set to 3, that is RMB which should do nothing when done in-UI. In either case this is a hotkey-only code path, so I don't think the chosen button actually changes the outcome.

Reply:
Regarding all the SetActiveCommand stuff, I decided to unify it with `(cmd, 1, true, false, Spring.GetModKeyState())`. In the engine that is: `bool SetActiveCommand(int cmdIndex, int button, bool leftMouseButton, bool rightMouseButton, bool alt, bool ctrl, bool meta, bool shift);`
From my testing this has no adverse effects on any click or hotkey behavior.